claimManagement/src/main/java/org/openimis/imisclaims/ClaimActivity.java

```diff
@@ -686,16 +686,7 @@ private boolean isValidData(){
     }
 
     private boolean isValidCHFID(){
-        //return true;
-	if (etCHFID.getText().toString().length() != 9) return false;
-	String chfid;
-	int Part1, Part2;
-	Part1 = Integer.parseInt(etCHFID.getText().toString())/10;
-	Part2 = Part1 % 7;
-
-	chfid = etCHFID.getText().toString().substring(0, 8) + Integer.toString(Part2);
-	return etCHFID.getText().toString().equals(chfid);
-
+        return true;
```

Comment:
call escape.CheckCHFID(etCHFID.getText().toString())

Reply:
Here we also need to call the escape procedure to check if valid chfid. We will keep only one place to update the validation and that will be the escape class.